lib/mixlib/install/generator/powershell/scripts/helpers.ps1.erb

```diff
@@ -112,21 +112,12 @@ function Test-ProjectPackage {
   param ($Path, $Algorithm = 'SHA256', $Hash)
 
   if (-not (get-command get-filehash -ErrorAction 'SilentlyContinue')) {
-    function disposable($o){($o -is [IDisposable]) -and (($o | get-member | foreach-object {$_.name}) -contains 'Dispose')}
-    function use($obj, [scriptblock]$sb){try {& $sb} catch [exception]{throw $_} finally {if (disposable $obj) {$obj.Dispose()}} }
-    function Get-FileHash ($Path, $Algorithm) {
-      $Path = (resolve-path $path).providerpath
-      $hash = @{Algorithm = $Algorithm; Path = $Path}
-      use ($c = Get-SHA256Converter) {
-        use ($in = (gi $path).OpenRead()) {
-          $hash.Hash = ([BitConverter]::ToString($c.ComputeHash($in))).Replace("-", "").ToUpper()
-        }
-      }
-      new-object PSObject -Property $hash
-    }
+    $DesiredHash = Get-FileHash $Path $Algorithm
+    new-object PSObject -Property $DesiredHash
   }
   Write-Verbose "Testing the $Algorithm hash for $path."
   $ActualHash = (Get-FileHash -Algorithm $Algorithm -Path $Path).Hash.ToLower()
+
   Write-Verbose "`tDesired Hash - '$hash'"
```

Comment:
You'll want to change `$hash` here to `$DesiredHash` as well as two lines down where we compare them.

I wonder about how `$Hash` is taken as a parameter to `Test-ProjectPackage`. Maybe that's cruft? The old code doesn't appear to really support that from a quick glance.